examples/vision/3D_image_classification.py

```diff
@@ -0,0 +1,454 @@
+
+
+def process_scan(path):
+    # read scan
+    volume = read_nifti_file(path)
+    # resize width and height
+    volume = resize_slices(volume)
+    # resize across z-axis
+    volume = resize_depth(volume)
+    return volume
+
+
+"""
+Let's read the paths of the CT scans from the class directories
+"""
+
+# folder "1" consist of right CT scans
+right_scan_paths = [
+    os.path.join(os.getcwd(), "NSCLC-Radiomics-Genomics/1", x)
+    for x in os.listdir("NSCLC-Radiomics-Genomics/1")
+]
+# folder "2" consist of left CT scans
+left_scan_paths = [
+    os.path.join(os.getcwd(), "NSCLC-Radiomics-Genomics/2", x)
+    for x in os.listdir("NSCLC-Radiomics-Genomics/2")
+]
+
+print("CT scans with cancerous regions in left side: " + str(len(left_scan_paths)))
+print("CT scans with cancerous regions in right side: " + str(len(right_scan_paths)))
+
+"""
+Let's visualize a CT scan and also look at it's dimensions.
+"""
+
+import matplotlib.pyplot as plt
+
+img = read_nifti_file(right_scan_paths[2])
+print("Dimension of the CT scan is:", img.shape)
+plt.imshow(img[:, :, 5], cmap="gray")
+
+"""
+Since a CT scan has many slices, let's visualize a montage of the slices.
+"""
+
+
+def plot_slices(data):
+    # plot a montage of 20 CT slices
+    a, b = 2, 10
+    data = np.rot90(np.array(data))
+    data = np.transpose(data)
+    data = np.reshape(data, (a, b, 512, 512))
+    test_data = data
+    r, c = test_data.shape[0], test_data.shape[1]
+
+    heights = [a[0].shape[0] for a in test_data]
+    widths = [a.shape[1] for a in test_data[0]]
+
+    fig_width = 12.0
+    fig_height = fig_width * sum(heights) / sum(widths)
+
+    f, axarr = plt.subplots(
+        r, c, figsize=(fig_width, fig_height), gridspec_kw={"height_ratios": heights}
+    )
+
+    for i in range(r):
+        for j in range(c):
+            axarr[i, j].imshow(test_data[i][j], cmap="gray")
+            axarr[i, j].axis("off")
+
+    plt.subplots_adjust(wspace=0, hspace=0, left=0, right=1, bottom=0, top=1)
+    plt.show()
+
+
+plot_slices(img[:, :, :20])
+
+"""
+## Build train and test datasets
+
+Read the scans from the class directories and assign labels.
+Lastly, split the dataset into train and test subsets.
+"""
+
+# read and process the scans
+right_scans = np.array([process_scan(path) for path in right_scan_paths])
+left_scans = np.array([process_scan(path) for path in left_scan_paths])
+
+# for the CT scans having cancerous regions in the right side
+# assign 1, similarly for left assign 0.
+right_labels = np.array([1 for _ in range(len(right_scans))])
+left_labels = np.array([0 for _ in range(len(left_scans))])
+
+# merge classes
+inputs = np.concatenate((right_scans, left_scans), axis=0)
+labels = np.concatenate((right_labels, left_labels), axis=0)
+labels = keras.utils.to_categorical(labels, 2)
+print("Size of raw data and labels:", inputs.shape, labels.shape)
+
+# split data in the ratio 70-30 for training and testing
+from sklearn.model_selection import train_test_split
```

Comment:
You don't need sklearn for this, you can just do it by hand

Reply:
Done!